OpenRA.Mods.Common/ActorExts.cs

```diff
@@ -141,6 +141,10 @@ public static void NotifyBlocker(this Actor self, IEnumerable<CPos> positions)
 					return false;
 			}
 
+			// Occupied by any actor, including ally/friendly units
+			if (self.World.ActorMap.GetActorsAt(pos).Any())
```

Comment:
You have pointed out some of the complexities involved, and we already have a large chunk of code in and around `MobileInfo.CanEnterCell` to deal with just this.  It would be a very bad idea to ignore that and write a *different* set of logic just for harvesters.